### PR TITLE
Support importing Wheels by specifying artifact classifiers

### PR DESCRIPTION
**Pull request by qlan2**
Hey guys,

I am qlan from Nerv team in Tableau Software, Inc.

Python Wheels is becoming more and more popular, but currently pivy-importer only supports importing sdist, this change will make it support importing wheels (and add ivy metadata) as well, and import all the runtime dependencies in Sdist format. This work has been filed at https://github.com/linkedin/pygradle/issues/186.

We used to pull sdist dependencies through
java -jar pivy-importer-0.6.17-all.jar --repo ivy-repo psycopg2_binary:2.7.4 Mako:1.0.7

with my changes in this pull request, we can pull both sdist (name:version) and wheels(name:version:classifier) through
java -jar pivy-importer-0.6.17-all.jar --repo ivy-repo psycopg2_binary:2.7.4:cp35-cp35m-win_amd64 psycopg2_binary:2.7.4 Mako:1.0.7

Run it in my local environment:
qlan@QLAN2 C:\Users\qlan\Downloads\test
$ java -jar pivy-importer-0.8.3-SNAPSHOT-all.jar --repo ivy-repo psycopg2_binary:2.7.4:cp35-cp35m-win_amd64 psycopg2_binary:2.7.4 Mako:1.0.7 --latest --lenient
15:47:20.924 INFO  c.l.p.importer.deps.WheelsDownloader - Pulling in psycopg2_binary:2.7.4:cp35-cp35m-win_amd64
15:47:21.328 INFO  c.l.p.importer.deps.SdistDownloader - Pulling in psycopg2-binary:2.7.4
15:47:21.434 INFO  c.l.p.importer.deps.SdistDownloader - Pulling in Mako:1.0.7
15:47:21.514 INFO  c.l.p.importer.deps.SdistDownloader - Pulling in MarkupSafe:1.0
15:47:21.519 INFO  c.l.python.importer.ImporterCLI - Execution Finished!

We also added some tests.
Please let me know if you have any question or feedback. Thanks!

### CODE REVIEW

**Comment by kopy07** [comment]
Hey @qlan2, how would a user specify the wheel?

**Comment by kopy07** [comment]
@qlan2 Awesome! Does the other PR depend on this one?

Can you add some rudimentary tests? I have a PR that completely re-writes the pivy-importer and it has a lot of tests. If you add them, I'll include them in my PR so when it lands we'll keep the same functionality.

**Comment by qlan2** [comment]
@kopy07 Yep, another PR includes the new functionality to pull down runtime dependencies of wheels and some useful tests, that PR depends on this one and currently is in the code review process within my team, I can push all the changes to this PR once my changes pass the code review. Thanks for your considerations!

**Comment by qlan2** [comment]
@kopy07 Hey Ethan, I just merged the changes of pulling down wheel's runtime dependencies and tests to this PR, and now this PR has full wheel support, please take a look at it, and let me know if you have any feedback. Thanks!

File: pivy-importer/build.gradle [hunk 1]
```diff
@@ -23,6 +23,12 @@ dependencies {
     compile 'ch.qos.logback:logback-classic:1.1.3'
     compile 'org.apache.commons:commons-compress:1.10'
     compile 'commons-io:commons-io:2.4'
+
+    testCompile group: "junit", name: "junit", version: "4.+"
+    testCompile group: 'org.mockito', name: 'mockito-core', version: '2.+'
+    testCompile ('org.spockframework:spock-core:1.1-groovy-2.4') {
+        exclude module: 'groovy-all' // Avoid conflict with version of Groovy loaded by Gradle
```
Comment (kopy07) [inline pivy-importer/build.gradle]: I don't think you need the exclude since we're not including `gadleApi()` or `groovyLocal()` here.

Comment (qlan2) [reply pivy-importer/build.gradle]: Good point, removed!

**Comment by qlan2** [comment]
@kopy07 Sure, thanks for your review! I have fixed most style checks, as for the license check for the resource files, I think it should be disabled since they are just resource files for testing purpose. You might merge my changes as "squash and merge", in this way, all the new commits can be squashed into one commit as "add wheel support", see details at [here](https://help.github.com/articles/about-pull-request-merges/#squash-and-merge-your-pull-request-commits).

**Comment by kopy07** [comment]
Yeah they can be pretty annoying. You can disable it on src/[test,integTest]/resource if you want.

Will use the merge + squash when CI passes.

**Comment by qlan2** [comment]
Cool sounds good, I made it only check groovy and java files. Hopefully it can pass the license check.

**Comment by kopy07** [comment]
@qlan2 It looks like if you do 

```
diff --git a/pivy-importer/src/test/groovy/com/linkedin/python/importer/distribution/PythonPackageTest.groovy b/pivy-importer/src/test/groovy/com/linkedin/python/importer/distribution/PythonPackageTest.gr
index 6d8a25c..2432ef2 100644
--- a/pivy-importer/src/test/groovy/com/linkedin/python/importer/distribution/PythonPackageTest.groovy
+++ b/pivy-importer/src/test/groovy/com/linkedin/python/importer/distribution/PythonPackageTest.groovy
@@ -49,8 +49,8 @@ class PythonPackageTest extends Specification {
         String testFileName = "PKG-INFO"
         String testEntryName = "WMI-1.4.8/$testFileName"
         when:
-        String actualText = testPythonPackage.explodeZipForTargetEntry(testEntryName).replaceAll(System.lineSeparator(), "\n")
-        String expectedText = new File(testDirectory, "$testFileName").text.replaceAll(System.lineSeparator(), "\n")
+        String actualText = testPythonPackage.explodeZipForTargetEntry(testEntryName).readLines()
+        String expectedText = new File(testDirectory, "$testFileName").text.readLines()
         then:
         actualText == expectedText
     }
```

That will fix the test.

**Comment by qlan2** [comment]
@kopy07 Thanks! I pushed another fix to replace "\r\n" with "\n" works for both platforms, but seems Github is down, the PR didn't get my latest commits. I will check later, if it still doesn't work, I will try your solution.

**Pull request merged**